### ISSUE TITLE
Improve documentation on creating a new user from Golang

### ISSUE THREAD

**Issue by erikd234** [pocketbase/pocketbase#5322]
TLDR: Request to add a code snippet about adding new Auth record for creating IE creating a user to the documentation.

I was writing tests and I wanted my test app to create a new test user each time I ran test pragmatically rather than having to go into the UI to do it. 

I read through all the documentation and spent lots of time reading the source code until I finally came up with.

```  
 // add a test user
    collection, err := dao.FindCollectionByNameOrId("users")
    if err != nil {
        panic("error retreeiving users collection, " + err.Error())
    }
    record := models.NewRecord(collection)

    form := forms.NewRecordUpsert(testApp, record)
    form.Email = TestUserEmail
    form.Password = TestUserPwd
    form.PasswordConfirm = TestUserPwd

    if err := form.Submit(); err != nil {
        panic("form failed to submit +" + err.Error())
    }
   ```
   
   This seems to work for my purposes of adding a test user to the testApp when I spin it up. 
   
   
 It would be nice if we could mark this in the docs since this took me too long to find out how to do (subjective, took about an hour,). I got a little confused since  in the db we just store the passwordHash, so i was trying to find out if I need to hash it myself when adding the record via go, or there is a thing like above that does the hashing for me (which is much better)
 
 Great work with pocketbase! I think its the best out of the box setup for using SQLITE out there I am loving it so far,  especially the integration tests api! 
 
 Keep it up!

**Comment by ganigeorgiev** [comment]
It won't be needed. This is simplified with the [refactoring](https://github.com/pocketbase/pocketbase/discussions/4355) and in the next upcoming v0.23 release (_there are no ETAs yet_) all system fields (id, password, email, etc.) are regular collection fields and you can create auth records just by doing:

```go
record := models.NewRecord(collection)
record.Set("password", "1234567890")    // or the designated helper SetPassword
record.Set("email", "test@example.com") // or the designated helper SetEmail
app.Save(record) // or app.SaveNoValidate(record) to skip fields validation
```

At the moment creating an auth record using the `RecordUpsert` is correct when you want validation. If you want a direct DB insert you have to save the record using the `app.Dao()` as shown in https://pocketbase.io/docs/go-migrations/#creating-new-auth-record.
